src/workerd/io/io-context.h

```diff
@@ -91,7 +91,8 @@ class IoContext_IncomingRequest {
   // This method is also used by some custom event handlers (see WorkerInterface::CustomEvent) that
   // need similar behavior, as well as the test handler. TODO(cleanup): Rename to something more
   // generic?
-  kj::Promise<bool> finishScheduled();
+  enum class FinishScheduledResult { COMPLETED, ABORTED, TIMEOUT };
+  kj::Promise<FinishScheduledResult> finishScheduled();
```

Comment:
Changing the return value here will also need a parallel change to the internal code since there are a couple callers of this method.

Reply:
👍 I've got that all wired up, just haven't put it up for PR yet.